lib/hcrypto/evp.c

```diff
@@ -857,60 +871,72 @@ EVP_CipherUpdate(EVP_CIPHER_CTX *ctx, void *out, int *outlen,
 
     *outlen = 0;
 
-    /**
-     * If there in no spare bytes in the left from last Update and the
-     * input length is on the block boundery, the EVP_CipherUpdate()
-     * function can take a shortcut (and preformance gain) and
-     * directly encrypt the data, otherwise we hav to fix it up and
-     * store extra it the EVP_CIPHER_CTX.
+    /*
+     * If there in no bytes left over from the last Update and the
+     * input length is on a block boundary, then we can take a
+     * shortcut (and preformance gain) and directly encrypt the
+     * data.
      */
-    if (ctx->buf_len == 0 && (inlen & ctx->block_mask) == 0) {
+    if (ctx->buf_len == 0 && inlen && (inlen & ctx->block_mask) == 0) {
+        /*
+         * OpenSSL's EVP_Cipher() returns the number of bytes on success, 0 on
+         * failure.
```

Comment:
EVP_Cipher calls ctx->cipher->do_cipher().   EVP_Cipher cannot be assigned to do_cipher.    do_cipher() functions return 1 for success and any other value is an error.

Reply:
Ah yes, I _fixed_ that harder in `cipher_do_cipher()`, so I can take this change out.